### DISCUSSION TITLE
Vue Support

### DISCUSSION THREAD

**Discussion by Ibz786** [crxjs/chrome-extension-tools#392] [Q&A] [answered]
Hi 👋🏽 

I just wanted to ask, does this plugin have Vue support? I've tried to create an extension using:
`vite@latest` and `npm i @crxjs/vite-plugin -D`

This is my `vite.config.js`
```js
import { fileURLToPath, URL } from 'url'

import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import { crx } from '@crxjs/vite-plugin'
import manifest from './manifest.json'

// https://vitejs.dev/config/
export default defineConfig({
  	plugins: [vue(), crx({manifest})],
  	resolve: {
    	    alias: {
      		'@': fileURLToPath(new URL('./src', import.meta.url))
    	    }
        }
});
```

But I'm getting the following error: `Uncaught TypeError: Failed to resolve module specifier "vue". Relative references must start with either "/", "./", or "../".` 

Not sure if I'm doing something wrong

Thanks 😄

**Comment by jacksteamdev** [comment, marked as answer]
Your config looks good to me! Hard to say what's wrong without a full repo to look at. 🙏 

Vue does work. You can check out some of the tests for basic Vue usage:

- [Vue Content Script](https://github.com/crxjs/chrome-extension-tools/tree/main/packages/vite-plugin/tests/e2e/mv3-vite-vue-content-script)
- [Vue Extension Page](https://github.com/crxjs/chrome-extension-tools/tree/main/packages/vite-plugin/tests/e2e/mv3-vite-vue-page)

**Reply by Ibz786** [reply]
Ok I'm not completely sure what I did, but I got it working 😆 thanks! 

I'll post up an example of my work just as an FYI for others in case. Thanks again! 👍🏽

**Reply by Ibz786** [reply]
I've added a repo here: [https://github.com/Ibz786/vue-vite-chrome-extension-template](https://github.com/Ibz786/vue-vite-chrome-extension-template) 

If it's helpful for your examples, please feel free to use / include it. The example I've got working also includes:
**Vue Router** and **Pinia**

**Reply by jacksteamdev** [reply]
Thanks for the [great article](https://dev.to/ibz786/create-a-vite-vue-chrome-extension-h3b) on DEV! 🙌 

I love the idea of using a router in a Chrome Extension, I'd be interested to hear your experience once you've finished your project. 🙏 I've seen people use hashes in the manifest, e.g., `options_page: 'index.html#options`

**Reply by Ibz786** [reply]
Hey @jacksteamdev  

Thanks much appreciated 😄🙌

I've been playing around with Vue, Vue-Router and your plugin and got routing working pretty well so far. Have added [another article on DEV](https://dev.to/ibz786/adding-routing-to-a-vue-vite-chrome-extension-46lm)

The solution I posted offers a cleaner URL without the need for hashing. Also updated my [example repo too](https://github.com/Ibz786/vue-vite-chrome-extension-template) 

The plugin is such a dream honestly, keep up the great work!

**Reply by Ibz786** [reply]
An update on the routing aspect, if anyone is curious, I got a better and more simpler way (really don't know why I didn't use this earlier 😅🤦‍♂️)

Simply do the following: 
```js
routes: [
  {
    path: '/',
    name: 'home',
    component: HomeView
  },
  {
    path: 'index.html',
    redirect: '/'
  },
...
]
```
Honestly, the Vue documentation is a blessing, I just missed it earlier: https://router.vuejs.org/guide/essentials/redirect-and-alias.html#redirect-and-alias. Happy dev'ing all 🤓

**Reply by joaomede** [reply]
Thank you [jacksteamdev](https://github.com/jacksteamdev).

Thank you [Ibz786](https://github.com/Ibz786) for article https://dev.to/ibz786/adding-routing-to-a-vue-vite-chrome-extension-46lm
